libsolidity/lsp/LanguageServer.cpp

```diff
@@ -173,6 +196,23 @@ void LanguageServer::changeConfiguration(Json::Value const& _settings)
 	}
 }
 
+vector<boost::filesystem::path> LanguageServer::allSolidityFilesFromProject() const
+{
+	namespace fs = boost::filesystem;
+
+	std::vector<fs::path> collectedPaths{};
+
+	// We explicitly decided against including all files from include paths but leave the possibility
+	// open for a future PR to enable such a feature to be optionally enabled (default disabled).
+
+	auto directoryIterator = fs::recursive_directory_iterator(m_fileRepository.basePath(), fs::symlink_option::recurse);
+	for (fs::directory_entry const& dirEntry: directoryIterator)
+		if (dirEntry.path().extension() == ".sol")
```

Comment:
Maybe this will also include directories.

Reply:
we tested that and concluded that it'll yield all kinds of directory entries. So I added an additional check to that if-clause.